frontend/app/src/components/staking/eth/EthStakingPage.vue

```diff
@@ -48,20 +59,30 @@ const premium = usePremium();
 const { t } = useI18n();
 
 async function refresh(userInitiated = false): Promise<void> {
-  const refreshValidators = async () => {
+  const refreshValidators = async (userInitiated: boolean) => {
     await fetchBlockchainBalances({
-      ignoreCache: true,
+      ignoreCache: userInitiated,
       blockchain: Blockchain.ETH2,
     });
     await accountsStore.fetchEth2Validators();
   };
 
-  const validatorRefresh: Promise<void>[] = (userInitiated ? [refreshValidators()] : []);
+  const updatePerformance = async (userInitiated = false): Promise<void> => {
+    await refreshPerformance(userInitiated);
+    // if the number of validators is bigger than the total entries in performance
+    // force a refresh of performance to pick the missing performance entries.
+    const totalValidators = get(eth2Validators).entriesFound;
+    const totalPerformanceEntries = get(performance).entriesTotal;
+    if (totalValidators > totalPerformanceEntries) {
```

Comment:
if there won't be a case where `totalValidators < totalPerformanceEntries` is truthy, then it is all good to me.

Reply:
Unless you manually go an remove the validators I don't think there would be a scenario where this could happen automatically. This is supposed to handle an edge case where the missing validators don't make it to performance due to timing.